examples/bigquery_user_info_updater_tool/README.md

```diff
@@ -0,0 +1,317 @@
+# BigQuery User Info Updater
+
+## Background
+BigQuery is often one of the preferred products for customers interested in 
+building a data lake on GCP. In some cases, product limitations can impact customers 
+who want to use BigQuery in their architecture. 
+
+## Motivating Example
+For example, imagine that you need 
+to ingest user information data in a table, but when an update is made, only a 
+partial row is ingested. In other words, only updated columns in the newly ingested 
+row will contain values and all other columns will be null. If you want to query only
+ the most up to date, full record, the solution could easily become complicated and inefficient.
+
+In the case mentioned above, imagine we have a destination table called
+`user_info_updates`,  which is populated by a Dataflow pipeline that gathers and
+ingests data from users registering or updating their profile on a website. 
+When users register for the first time, an entire row is ingested in the table:
+##### user_info_updates 
+
+ | user_id      | ingestTimestamp    | attribute1     |attribute2    |attribute3     |
+| :------------- | :----------- | :----------- | :----------- |  :----------- | 
+|  userA | 1   | xyz  |abc  |qwerty|
+|  userB | 2   | poc  |mvp  |ytrewq|
+|  userC | 3   | zyx  |pvm  |qywter|
+
+Now imagine that userA and userB update their information. In this case the 
+data source will send only a partial updated row per update: 
+
+
+##### user_info_updates 
+
+ user_id      | ingestTimestamp    | attribute1     |attribute2    |attribute3     |
+| :------------- | :----------- | :----------- | :----------- |  :----------- | 
+|  userA | 1   | xyz  |abc  |qwerty|
+|  userB | 2   | poc  |mvp  |ytrewq|
+|  userC | 3   | zyx  |pvm  |qywter|
+|  userA | 4   |   |ghj  | |
+|  userA | 5   |   |  |lot|
+|  userB | 6   |  cor |  | |
+|  userA | 7   |   | dat ||
+
+These partial rows are not very useful since any non-updated values are missing. 
+To provide analysts/data scientists/etc with the latest, full record for each user,
+ a table called user_info_final can be created using a series of SQL queries: 
+
+##### user_info_final
+
+ user_id      | ingestTimestamp    | attribute1     |attribute2    |attribute3     |
+| :------------- | :----------- | :----------- | :----------- |  :----------- | 
+|  userA | 7   | xyz  |dat  |lot|
+|  userB | 6   | cor  |mvp  |ytrewq|
+|  userC | 3   | zyx  |pvm  |qywter|
+
+The BigQuery User Info Updater to can be used to solve this problem with series of advanced
```

Comment:
jaketf on Aug 27, 2019 Member
| The BigQuery User Info Updater to can be used to solve this problem with serious of advanced
-- | --
  | The BigQuery User Info Updater to can be used to solve this problem with a series of advanced